redis_commands.c

```diff
@@ -2548,6 +2548,54 @@ int redis_hsetnx_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
         cmd, cmd_len, slot);
 }
 
+int
+redis_hrandfield_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
+                      char **cmd, int *cmd_len, short *slot, void **ctx)
+{
+    char *key;
+    int count = 0;
+    size_t key_len;
+    smart_string cmdstr = {0};
+    zend_bool withvalues = 0;
+    zval *z_opts = NULL, *z_ele;
+
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "s|a",
+                              &key, &key_len, &z_opts) == FAILURE)
+    {
+        return FAILURE;
+    }
+
+    if (z_opts && Z_TYPE_P(z_opts) == IS_ARRAY) {
```

Comment:
I think we know `Z_TYPE_P(z_opts) == IS_ARRAY` since we required one?

Reply:
Make sense, but before removing this check I will try how will it work with passed by reference variable